src/main/java/org/jetbrains/kotlinx/lincheck/runner/ParallelThreadsRunner.kt

```diff
@@ -49,7 +49,7 @@ open class ParallelThreadsRunner(
     waits: List<IntArray>?
 ) : Runner(strategy, testClass, validationFunctions) {
     private lateinit var testInstance: Any
-    private val executor = newFixedThreadPool(scenario.threads, ParallelThreadsRunner::TestThread)
+    private val executor = newFixedThreadPool(scenario.threads, { TestThread(it, this) })
 
```

Comment:
Let's add a field `testThreadHash` initialized with `this.hasCode()` or `Random.nextInt()`

Reply:
Let's also add comments on these changes